messages/teleporter/message_manager.go

```diff
@@ -87,13 +109,59 @@ func (m *messageManager) ShouldSendMessage(warpMessageInfo *vmtypes.WarpMessageI
 	if !ok {
 		return false, fmt.Errorf("relayer not configured to deliver to destination. destinationChainID=%s", destinationChainID.String())
 	}
-	if !destinationClient.Allowed(destinationChainID, teleporterMessage.AllowedRelayerAddresses) {
+	senderAddress := destinationClient.SenderAddress()
+	if !isAllowedRelayer(teleporterMessage.AllowedRelayerAddresses, senderAddress) {
+		m.logger.Info("Relayer EOA not allowed to deliver this message.")
+		return false, nil
+	}
+	if !isDestination(destinationChainID, destinationClient.DestinationChainID()) {
```

Comment:
This check seems redundant given that we look up the destination client by the chain ID already. If we choose to keep it, could we move this to right after the `destinationClient` is looked up from the `destinationClients` map?

Reply:
Agreed. I've removed it altogether, but left the corresponding method in `DestinationClient`, as it makes sense as a getter for that interface.